### ISSUE TITLE
Add support for `@import`

### ISSUE THREAD

**Issue by Arkellys** [gajus/eslint-plugin-jsdoc#1226]
Clone of [this feature request](https://github.com/jsdoc-type-pratt-parser/jsdoc-type-pratt-parser/issues/165), originally posted on `jsdoc-type-pratt-parser`.

## Motivation

Typescript v5.5 introduces the tag [`@import`](https://devblogs.microsoft.com/typescript/announcing-typescript-5-5-beta/#type-imports-in-jsdoc), which allows importing types (or other stuff) within JSDoc:

```js
/** @import { SomeType } from "some-module" */
```

Would it be possible to add support for this new syntax?

## Current behavior

With `mode: "typescript"`

**types.js**
```js
export const allowImports = true; // Trick to allow importing

/** @typedef {string} TestOne */
/** @typedef {string} TestTwo */
```

**other.js**
```js
/**
 * @import { TestOne, TestTwo } from "./types"  🡒  Syntax error in type: TestOne, TestTwo eslint(jsdoc/valid-types)
 * @import { TestOne } from "./types"  🡒  The type 'TestOne' is undefined. eslint(jsdoc/no-undefined-types)
 * @import * as Types from "./types"  🡒  No error
 */

// And regardless of the import used: 

/** 
 * @typedef {TestOne} Test  🡒  The type 'TestOne' is undefined. eslint(jsdoc/no-undefined-types)
 */ 
```

Except for the errors, the imports works well.
## Desired behavior

Being able to use `@import` without errors.

**Comment by brettz9** [comment]
To replicate, I will need your full ESLint config and details (e.g., version of `eslint-plugin-jsdoc`).

**Comment by Arkellys** [comment]
> To replicate, I will need your full ESLint config and details (e.g., version of `eslint-plugin-jsdoc`).

Oh yes sorry, I put it on the other issue and I forgot to add it here, my full config + version used is available on [this repository](https://github.com/Arkellys/eslint-config-arklint). The JSDoc config specifically is on [this file](https://github.com/Arkellys/eslint-config-arklint/blob/main/extensions/jsdoc.js).

**Comment by brettz9** [comment]
The `import` issue should be fixed in v48.2.3.

**Comment by brettz9** [comment]
Sorry, got confused. It doesn't fix it.

**Comment by brettz9** [comment]
The `no-undefined-types` rule is currently discouraged if you are using TypeScript checking (which checks for undefined types anyways). If you change your config to "recommended-typescript-flavor" or "recommended-typescript-flavor-error" (for JavaScript; if using TypeScript syntax, use "recommended-typescript" or "recommended-typescript-error"), this rule will be disabled and you will not see the error.

**Comment by Arkellys** [comment]
All right, thank you for the information, and the fix! I'll see about adjusting my configuration. 🙂
